### PR TITLE
feat(content-conformance): add GH comment reporter

### PR DESCRIPTION
**Pull request by BRKalow**
🎟️ [Asana Task]()

---

## Description

Add a reporter to output formatted markdown intended to be posted as a GitHub PR comment.

## PR Checklist 🚀

- [ ] Conduct thorough self-review.
- [ ] Add or update tests as appropriate.
- [ ] Write a useful description (above) to give reviewers appropriate context.
- [ ] Identify (in the description above) and document (add Asana tasks on [this board](https://app.asana.com/0/1100423001970639/list)) any technical debt that you're aware of, but are not addressing as part of this PR.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1203331966445549

### CODE REVIEW

File: packages/content-conformance/src/reporters/markdown-reporter.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  const lines = ['### 📄 Content Checks', '']
+
+  // commit & timestamp
+
+  // TODO: the sha isn't available to the reporter yet
+  // lines.push(`Latest commit: ${sha} (updated: ${new Date().toUTCString()})`, '')
+
+  // Make the timestamp generic so we can snapshot the output
+  const updatedTimestamp =
+    process.env.NODE_ENV === 'test' ? 'Date' : new Date().toUTCString()
```
Comment (BRKalow) [inline packages/content-conformance/src/reporters/markdown-reporter.ts:18]: This ensures we can snapshot the output without having a timestamp present

File: packages/content-conformance/src/reporters/markdown-reporter.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+export function markdownReporter(files: VFile[] /* sha?: string */) {
+  // @ts-expect-error - something is wrong with the VFile type used in vfile-statistics
+  const stats = statistics(files)
+
+  const lines = ['### 📄 Content Checks', '']
+
+  // commit & timestamp
+
+  // TODO: the sha isn't available to the reporter yet
+  // lines.push(`Latest commit: ${sha} (updated: ${new Date().toUTCString()})`, '')
```
Comment (BRKalow) [inline packages/content-conformance/src/reporters/markdown-reporter.ts:14]: x-ref: https://app.asana.com/0/1100423001970639/1203535318725182/f

File: packages/content-conformance/src/reporters/markdown-reporter.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+      if (message.position) {
+        columns.push(`\`${stringifyPosition(message.position)}\``)
+      } else {
+        columns.push('')
+      }
+
+      columns.push(message.reason)
+
+      // TODO: link to descriptive page for the rule
+      columns.push(`[\`${message.ruleId}\`](#TODO)`)
```
Comment (BRKalow) [inline packages/content-conformance/src/reporters/markdown-reporter.ts:47]:
```suggestion
      columns.push(`\`${message.ruleId}\``)
```

File: packages/content-conformance/src/reporters/markdown-reporter.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+      lines.push(`|${columns.join('|')}|`, '')
+    })
+  })
+
+  // Footnote
+  // TODO: link to a README
+  lines.push(
+    '_Looking for more information? Check out the [content checks README](#TODO)_',
+    ''
+  )
```
Comment (BRKalow) [inline packages/content-conformance/src/reporters/markdown-reporter.ts:58]:
```suggestion
//  lines.push(
//    '_Looking for more information? Check out the [content checks README](#TODO)_',
//    ''
//  )
```

File: packages/content-conformance/src/reporters/markdown-reporter.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+
+  // commit & timestamp
+
+  // Make the timestamp generic so we can snapshot the output
+  const updatedTimestamp =
+    process.env.NODE_ENV === 'test' ? 'Date' : new Date().toUTCString()
+  lines.push(`Updated: ${updatedTimestamp}`, '')
+
+  // <details> header
+  lines.push(`<details><summary> Found ${stats.fatal} error(s)</summary>`, '')
```
Comment (thiskevinwang) [inline packages/content-conformance/src/reporters/markdown-reporter.ts:19]:
```suggestion
  lines.push(`<details><summary>Found ${stats.fatal} error(s)</summary>`, '')
```

File: packages/content-conformance/src/__tests__/runner.test.ts [hunk 6]
```diff
@@ -217,4 +217,53 @@ describe('ContentConformanceRunner', () => {
...
+    await runner.init()
+
+    await runner.run()
+
+    expect(await runner.report()).toMatchInlineSnapshot(`
+      "### 📄 Content Checks
+
+      Updated: Date
+
+      <details><summary> Found 4 error(s)</summary>
```
Comment (thiskevinwang) [inline packages/content-conformance/src/__tests__/runner.test.ts:238]:
```suggestion
      <details><summary>Found 4 error(s)</summary>
```

**Review by thiskevinwang — APPROVED** [review]
👏 
(1 tiny nit)

File: packages/content-conformance/src/__tests__/runner.test.ts [hunk 7]
```diff
@@ -217,4 +217,53 @@ describe('ContentConformanceRunner', () => {
...
+    const runner = new ContentConformanceRunner({
+      cwd: fixturePath,
+      reporter: 'markdown',
+    })
+
+    await runner.init()
+
+    await runner.run()
+
+    expect(await runner.report()).toMatchInlineSnapshot(`
```
Comment (BRKalow) [inline packages/content-conformance/src/__tests__/runner.test.ts:233]: rendered:

### 📄 Content Checks
Updated: Date
<details><summary>Found 4 error(s)</summary>

#### `content/has-frontmatter.mdx`
| Position | Description | Rule |
|---|---|---|
|`12:1-12:11`|Level 1 headings are not allowed|`local-no-h1`|

#### `content/index.mdx`
| Position | Description | Rule |
|---|---|---|
|`1:1-1:8`|Level 1 headings are not allowed|`local-no-h1`|

#### `content/no-h1.mdx`
| Position | Description | Rule |
|---|---|---|
|`1:1-1:27`|Must have a level 1 heading at the top of the file.|`must-have-h1`|

#### `content/nested/nested.mdx`
| Position | Description | Rule |
|---|---|---|
|`1:1-1:9`|Level 1 headings are not allowed|[`local-no-h1`|

</details>

**Pull request merged**